packages/shell-api/src/helpers.ts

```diff
@@ -487,33 +498,45 @@ export async function getPrintableShardStatus(
   ]);
   result.balancer = balancerRes;
 
-  const databases = await (await configDB.getCollection('databases').find())
-    .sort({ name: 1 })
-    .toArray();
-
+  // All databases in config.databases + those implicitly referenced
+  // by a sharded collection in config.collections
+  // (could become a single pipeline using $unionWith when we drop 4.2 server support)
+  const [databases, collections] = await Promise.all([
+    (async () =>
+      await (await configDB.getCollection('databases').find())
+        .sort({ _id: 1 })
+        .toArray())(),
+    (async () =>
+      await (
+        await configDB.getCollection('collections').find({
+          ...onlyShardedCollectionsInConfigFilter,
+        })
+      )
+        .sort({ _id: 1 })
+        .toArray())(),
+  ]);
   // Special case the config db, since it doesn't have a record in config.databases.
   databases.push({ _id: 'config', primary: 'config', partitioned: true });
+
+  for (const coll of collections) {
+    if (!databases.find((db) => coll._id.startsWith(db._id + '.'))) {
+      databases.push({ _id: coll._id.split('.')[0] });
+    }
+  }
+
   databases.sort((a: any, b: any): number => {
     return a._id.localeCompare(b._id);
   });
 
-  result.databases = await Promise.all(
-    databases.map(async (db) => {
-      const escapeRegex = (string: string): string => {
-        return string.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
-      };
-      const colls = await (
-        await configDB
-          .getCollection('collections')
-          .find({ _id: new RegExp('^' + escapeRegex(db._id) + '\\.') })
-      )
-        .sort({ _id: 1 })
-        .toArray();
+  result.databases = (
+    await Promise.all(
+      databases.map(async (db) => {
+        const colls = collections.filter((coll) =>
+          coll._id.startsWith(db._id + '.')
+        );
```

Comment:
If we need to get all collection infos anyway, we might as well avoid doing the extra query that we did _for each database_ here